### ISSUE TITLE
JSON parse gives wrong Float64 result

### ISSUE THREAD

**Issue by hanabi1224** [crystal-lang/crystal#10920]
Repro steps
Below code show difference between from_json and to_f64. (result of to_f64 is consistent with other languages while from_json is not)
```crystal
num = "43.474709000000132"
puts Float64.from_json(num)
# 43.474709000000125
puts num.to_f64
# 43.47470900000013
```

Crystal compiler version
> Crystal 1.0.0 [dd40a2442] (2021-03-22)
LLVM: 10.0.0
Default target: x86_64-unknown-linux-gnu

**Comment by HertzDevil** [comment]
Should be fixed by #10447

**Comment by straight-shoota** [comment]
Related to #10419
